### PR TITLE
Path cosmetic updates/paths

### LINKED ISSUE #2001: All paths use Boost objects

### PR DESCRIPTION
**Pull request by yying**
* Added cosmetic changes to file paths by converting path as strings to paths as boost::filesystem::path objects (in platform agnostic locations)
* Added support for test utils in Windows

Closes #2001

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

File: osquery/tests/windows/utils.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  // Python is installed here if provisioning script is used
+  python_path += "\\tools\python2\\python.exe";
+  if (::CreateProcessA(python_path.c_str(), &mutable_argv[0], NULL, NULL, FALSE,
+                       0, NULL, NULL, &si, &pi)) {
+    server.reset(new PlatformProcess(pi.hProcess));
+    ::CloseHandle(pi.hThread);
+    ::CloseHandle(pi.hProcess);
+  }
+
+  return std::move(server);
```
Comment (PoppySeedPlehzr) [inline osquery/tests/windows/utils.cpp]: Wonder if this line doesn't need the same as above.

**Comment by PoppySeedPlehzr** [comment]
Ok to test

**Comment by osqueryer** [comment]
Build origin/pr/2155/merge (Job: 2754) failed :-1:

**Comment by osqueryer** [comment]
Build origin/pr/2155/merge (Job: 2755) failed :-1:

**Comment by osqueryer** [comment]
Build origin/pr/2155/merge (Job: 2756) failed :-1:

**Comment by osqueryer** [comment]
Build origin/pr/2155/merge (Job: 2757) is good! :+1:

File: osquery/tests/test_util.cpp [hunk 2]
```diff
@@ -37,14 +37,17 @@ std::string kFakeDirectory = "";
 #ifdef DARWIN
 std::string kTestWorkingDirectory = "/private/tmp/osquery-tests";
 #else
-std::string kTestWorkingDirectory = "/tmp/osquery-tests";
+std::string kTestWorkingDirectory =
```
Comment (theopolis) [inline osquery/tests/test_util.cpp]: We're being inconsistent with strings and path objects within tests. Shouldn't we avoid complex initialization in global scope? Or at least try to be consistent.

**Pull request closed without merging**